src/aws/auth.rs

```diff
@@ -69,6 +69,21 @@ pub enum AwsAuthentication {
         /// The AWS secret access key.
         #[configurable(metadata(docs::examples = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"))]
         secret_access_key: SensitiveString,
+
+        /// The ARN of an [IAM role][iam_role] to assume.
+        ///
+        /// [iam_role]: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
+        #[configurable(metadata(docs::examples = "arn:aws:iam::123456789098:role/my_role"))]
+        assume_role: Option<String>,
+
+        /// The [AWS region][aws_region] to send STS requests to.
+        ///
+        /// If not set, this will default to the configured region
+        /// for the service itself.
+        ///
+        /// [aws_region]: https://docs.aws.amazon.com/general/latest/gr/rande.html#regional-endpoints
+        #[configurable(metadata(docs::examples = "us-west-2"))]
+        region: Option<String>,
```

Comment:
Wondering if it would make sense to call out that this is only relevant when `assume_role` is set.

Reply:
I have no strong opinion. It was parameter copy-pasta from one enum struct variant to another. Not sure what clarifying this would do though, because the same parameter (and its docs) are in multiple enum variants, but the docs generator would only pick one? It's relevant when `assume_role` is set here, but is relevant all the time if not using keys but using IMDS, etc.